src/FMBot.Bot/Services/IndexService.cs

```diff
@@ -193,26 +216,82 @@ public async Task<GuildUser> GetOrAddUserToGuild(Persistence.Domain.Models.Guild
             return guild.GuildUsers.First(f => f.UserId == user.UserId);
         }
 
-        public async Task UpdateUserName(IGuildUser discordGuildUser, int userId, int? guildId)
+        public async Task UpdateUser(IGuildUser discordGuildUser, int userId, int? guildId)
+        {
+            var discordName = discordGuildUser.Nickname ?? discordGuildUser.Username;
+
+            await using var db = this._contextFactory.CreateDbContext();
+
+            const string sql = "UPDATE guild_users " +
+                               "SET user_name =  @userName, " +
+                               "who_knows_whitelisted =  @whoKnowsWhitelisted " +
+                               "WHERE guild_id = @guildId AND user_id = @userId ";
+
+            DefaultTypeMap.MatchNamesWithUnderscores = true;
+            await using var connection = new NpgsqlConnection(this._botSettings.Database.ConnectionString);
+            await connection.OpenAsync();
+
+            var guild = (await connection.QueryAsync<Persistence.Domain.Models.Guild>("SELECT * FROM guilds WHERE guild_id = @guildId", new
+            {
+                guildId
+            }))
+            .ToList().FirstOrDefault();
```

Comment:
Instead of doing this I think there's a `QueryFirstOrDefaultAsync` method you can use